x-pack/plugins/cloud_integrations/cloud_experiments/public/launch_darkly_client/launch_darkly_client.ts

```diff
@@ -22,46 +23,81 @@ export interface LaunchDarklyUserMetadata
+  private loadingClient$ = new BehaviorSubject<boolean>(true);
+  private _launchDarklyClient$ = this.launchDarklyClient.asObservable();
 
   constructor(
     private readonly ldConfig: LaunchDarklyClientConfig,
     private readonly kibanaVersion: string
   ) {}
 
   public async updateUserMetadata(userMetadata: LaunchDarklyUserMetadata) {
+    if (this.canceled) return;
+
     const { userId, ...userMetadataWithoutUserId } = userMetadata;
     const launchDarklyUser: LDSingleKindContext = {
       ...userMetadataWithoutUserId,
       kind: 'user',
       key: userId,
     };
-    if (this.launchDarklyClient) {
-      await this.launchDarklyClient.identify(launchDarklyUser);
+
+    let launchDarklyClient: LDClient | null = null;
+    if (this.initiated) {
+      launchDarklyClient = await firstValueFrom(this.launchDarklyClient$);
+    }
+
+    if (launchDarklyClient) {
+      await launchDarklyClient.identify(launchDarklyUser);
     } else {
+      this.initiated = true;
       const { initialize, basicLogger } = await import('launchdarkly-js-client-sdk');
-      this.launchDarklyClient = initialize(this.ldConfig.client_id, launchDarklyUser, {
+      launchDarklyClient = initialize(this.ldConfig.client_id, launchDarklyUser, {
         application: { id: 'kibana-browser', version: this.kibanaVersion },
         logger: basicLogger({ level: this.ldConfig.client_log_level }),
       });
+      this.launchDarklyClient.next(launchDarklyClient);
+      this.loadingClient$.next(false);
     }
   }
 
   public async getVariation<Data>(configKey: string, defaultValue: Data): Promise<Data> {
-    if (!this.launchDarklyClient) return defaultValue; // Skip any action if no LD User is defined
-    await this.launchDarklyClient.waitForInitialization();
-    return await this.launchDarklyClient.variation(configKey, defaultValue);
+    const launchDarklyClient = await firstValueFrom(this.launchDarklyClient$);
+    if (!launchDarklyClient) return defaultValue; // Skip any action if no LD User is defined
+    await launchDarklyClient.waitForInitialization();
+    return await launchDarklyClient.variation(configKey, defaultValue);
   }
 
   public reportMetric(metricName: string, meta?: unknown, value?: number): void {
-    if (!this.launchDarklyClient) return; // Skip any action if no LD User is defined
-    this.launchDarklyClient.track(metricName, meta, value);
+    firstValueFrom(this.launchDarklyClient$).then((launchDarklyClient) => {
+      if (!launchDarklyClient) return; // Skip any action if no LD User is defined
+      launchDarklyClient.track(metricName, meta, value);
+    });
   }
 
   public stop() {
-    this.launchDarklyClient
-      ?.flush()
-      // eslint-disable-next-line no-console
-      .catch((err) => console.warn(err));
+    firstValueFrom(this.launchDarklyClient$).then((launchDarklyClient) => {
+      launchDarklyClient
+        ?.flush()
+        // eslint-disable-next-line no-console
+        .catch((err) => console.warn(err));
```

Comment:
Unrelated to this PR.
Nit: core already provides a logger in the browser. Can we use it instead of `console` so that we have one eslint-disable less? 😇

Reply:
Sure. 👍  Addressed in https://github.com/elastic/kibana/pull/181124/commits/d9eb81b2a6ae5983614ad9ae9cc30e9f820f2e38